demisto_sdk/commands/download/downloader.py

```diff
@@ -0,0 +1,510 @@
+        :return: None
+        """
+        file_entity: str = custom_content_object['entity']
+        file_path: str = custom_content_object['path']
+        file_type: str = custom_content_object['type']
+        file_name: str = custom_content_object['name']
+
+        output_path: str = os.path.join(self.output_pack_path, file_entity)
+        # dir name should be the same as file name without separators mentioned in constants.py
+        dir_name: str = self.create_dir_name(file_name)
+        output_path: str = os.path.join(output_path, dir_name)
+
+        extractor = Extractor(input=file_path, output=output_path, file_type=file_type, base_name=dir_name,
+                              no_auto_create_dir=True, no_logging=True, no_pipenv=True)
+        extractor.extract_to_package_format()
+
+        print_color(f'- Added {file_type} "{file_name}"', LOG_COLORS.NATIVE)
+
+    def merge_new_file(self, custom_content_object: dict) -> None:
+        """
+        Merges new files of type playbook/json (not existing in the output pack)
+        :param custom_content_object: The custom content object to merge into the pack
+        :return: None
+        """
+        file_entity: str = custom_content_object['entity']
+        file_path: str = custom_content_object['path']
+        file_type: str = custom_content_object['type']
+        file_name: str = custom_content_object['name']
+
+        output_path: str = os.path.join(self.output_pack_path, file_entity)
+        output_name: str = os.path.basename(file_path)
+        output_path: str = os.path.join(output_path, output_name)
+        shutil.move(src=file_path, dst=output_path)
+
+        print_color(f'- Added {file_type} "{file_name}"', LOG_COLORS.NATIVE)
+
+    def get_corresponding_pack_content_object(self, custom_content_object: dict) -> dict:
+        """TODO: add unit test
+        Returns the corresponding pack content object to the given custom content object
+        :param custom_content_object: The custom content object to merge into the pack
+        :return: The corresponding pack content object
+        """
+        file_entity: str = custom_content_object['entity']
+        file_name: str = custom_content_object['name']
+        pack_entity_instances: list = self.pack_content[file_entity]
+
+        for pack_entity_instance in pack_entity_instances:
+            if file_name in pack_entity_instance:
+                return pack_entity_instance
+        return {}
+
+    @staticmethod
+    def get_corresponding_pack_file_object(searched_basename: str, pack_content_object: dict) -> dict:
+        """TODO: add unit test
+        Searches for the file named searched_basename under the pack content object and returns it
+        :param searched_basename: The basename to look for
+        :param pack_content_object: The pack content object
+        :return: The pack file object
+        """
+        for _, file_objects in pack_content_object.items():
```

Comment:
```suggestion
        for file_objects in pack_content_object.values():
```

Reply:
This will force us to use casting: `list(pack_content_object.values())[0]` which is ugly.

Reply:
Not sure I am following

Reply:
> @guyfreund seems that `dictor` does json query. Does it support full syntax of JSON query?
> http://www.jsonquerytool.com/

We don't use it with JSON query syntax. 
We use it to extract values from the data given a dot notation key. For example:
data = {'a': {'b': {'c': 1}}}
doctor(data, 'a.b.c') returns 1